probe-rs-cli-util/src/common_options.rs

```diff
@@ -0,0 +1,453 @@
+        let target = if let Some(chip_name) = &self.chip {
+            let target = probe_rs::config::get_target_by_name(chip_name).map_err(|error| {
+                OperationError::ChipNotFound {
+                    source: error,
+                    name: chip_name.clone(),
+                }
+            })?;
+
+            TargetSelector::Specified(target)
+        } else {
+            TargetSelector::Auto
+        };
+
+        Ok(target)
+    }
+
+    /// Attaches to specified probe and configures it.
+    pub fn attach_probe(&self) -> Result<Probe, OperationError> {
+        let mut probe = {
+            if self.dry_run {
+                Probe::from_specific_probe(Box::new(FakeProbe::new()));
+            }
+
+            // If we got a probe selector as an argument, open the probe
+            // matching the selector if possible.
+            match &self.probe_selector {
+                Some(selector) => {
+                    Probe::open(selector.clone()).map_err(OperationError::FailedToOpenProbe)
+                }
+                None => {
+                    // Only automatically select a probe if there is
+                    // only a single probe detected.
+                    let list = Probe::list_all();
+                    if list.len() > 1 {
+                        return Err(OperationError::MultipleProbesFound { number: list.len() });
+                    }
+
+                    if let Some(info) = list.first() {
+                        Probe::open(info).map_err(OperationError::FailedToOpenProbe)
+                    } else {
+                        Err(OperationError::NoProbesFound)
+                    }
+                }
+            }
+        }?;
+
+        // Select protocol and speed
+        probe.select_protocol(self.protocol).map_err(|error| {
+            OperationError::FailedToSelectProtocol {
+                source: error,
+                protocol: self.protocol,
+            }
+        })?;
+        if let Some(speed) = self.speed {
+            let _actual_speed = probe.set_speed(speed).map_err(|error| {
+                OperationError::FailedToSelectProtocolSpeed {
+                    source: error,
+                    speed,
+                }
+            })?;
+        }
+
+        Ok(probe)
+    }
+
+    pub fn attach_session(
+        &self,
+        probe: Probe,
+        target: TargetSelector,
+    ) -> Result<Session, OperationError> {
+        let session = if self.connect_under_reset {
+            probe.attach_under_reset(target)
+        } else {
+            probe.attach(target)
+        }
+        .map_err(|error| OperationError::AttachingFailed {
+            source: error,
+            connect_under_reset: self.connect_under_reset,
+        })?;
+
+        Ok(session)
+    }
+
+    /// Convenience method that attaches to the specified probe, target,
+    /// and target session.
+    pub fn create_session(&self) -> Result<Session, OperationError> {
```

Comment:
Rename to `simple_attach`.

Reply:
This function may as well also call `maybe_load_chip_desc` and `early_exit`. Return `Option<Session>` instead. On `None`, `early_exit` returned true and program should exit.

Reply:
`early_exit` is defined in `FlashOptions`, so we cannot call it. Changing the name suffices.